bundles/org.openhab.binding.teslapowerwall/src/main/java/org/openhab/binding/teslapowerwall/internal/TeslaPowerwallWebTargets.java

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.teslapowerwall.internal;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.teslapowerwall.internal.api.BatterySOE;
+import org.openhab.binding.teslapowerwall.internal.api.GridStatus;
+import org.openhab.binding.teslapowerwall.internal.api.MeterAggregates;
+import org.openhab.binding.teslapowerwall.internal.api.Operations;
+import org.openhab.binding.teslapowerwall.internal.api.SystemStatus;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Handles performing the actual HTTP requests for communicating with Tesla Powerwall units.
+ *
+ * @author Paul Smedley - Initial Contribution
+ *
+ */
+@NonNullByDefault
+public class TeslaPowerwallWebTargets {
+    private static final int TIMEOUT_MS = 30000;
+
+    private String getBatterySOEUri;
+    private String getGridStatusUri;
+    private String getSystemStatusUri;
+    private String getMeterAggregatesUri;
+    private String getTokenUri;
+    private String getOperationUri;
+    private String setOperationUri;
+    private final Logger logger = LoggerFactory.getLogger(TeslaPowerwallWebTargets.class);
+
+    public TeslaPowerwallWebTargets(String ipAddress) {
+        String baseUri = "https://" + ipAddress + "/";
+        getBatterySOEUri = baseUri + "api/system_status/soe";
+        getGridStatusUri = baseUri + "api/system_status/grid_status";
+        getSystemStatusUri = baseUri + "api/system_status";
+        getMeterAggregatesUri = baseUri + "api/meters/aggregates";
+        getTokenUri = baseUri + "api/login/Basic";
+        getOperationUri = baseUri + "api/operation";
+        setOperationUri = baseUri + "api/operation";
+    }
+
+    public BatterySOE getBatterySOE(String token) throws TeslaPowerwallCommunicationException {
+        String response;
+        Properties headers = new Properties();
+        headers.setProperty("Content-Type", token);
+
+        synchronized (this) {
+            try {
+                response = HttpUtil.executeUrl("GET", getBatterySOEUri, headers, null, null, TIMEOUT_MS);
+            } catch (IOException ex) {
```

Comment:
some excpetions missing like interrupt and timeout.
also apply to other http calls in this binding.

Reply:
looking into this,  but can't get it working.  will keep trying tomorrow. I get errors like:
[ERROR] /home/psmedley/openhab-addons/bundles/org.openhab.binding.teslapowerwall/src/main/java/org/openhab/binding/teslapowerwall/internal/TeslaPowerwallWebTargets.java:[70,36] Unreachable catch block for java.lang.InterruptedException. This exception is never thrown from the try statement body
[ERROR] /home/psmedley/openhab-addons/bundles/org.openhab.binding.teslapowerwall/src/main/java/org/openhab/binding/teslapowerwall/internal/TeslaPowerwallWebTargets.java:[70,59] Unreachable catch block for java.util.concurrent.TimeoutException. This exception is never thrown from the try statement body

Reply:
Your code is correct, `HttpUtil.executeUrl` only throws `IOException`, the others mentioned are wrapped inside the utility class:

https://github.com/openhab/openhab-core/blob/5f1b708f27a07c9112bb8c23bbb8e7cb4370fcd6/bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/HttpUtil.java#L254-L265

Otherwise you would have compiler error, since these exceptions are checked exceptions.

Reply:
Thanks, resolving thsi comment.